File: api-elements/src/main/java/com/rainmore/cms/domains/system/SystemProperty.java
package com.rainmore.cms.domains.system;

import com.rainmore.cms.domains.HasId;
import com.rainmore.cms.domains.Nameable;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;

@Entity(name = "system.properties")
@Table(name = "systemProperties")
public class SystemProperty implements HasId<String>, Nameable {
    private String id;
    private String data;

    public SystemProperty() {
    }

    public SystemProperty(String id, String data) {
        this.id = id;
        this.data = data;
    }

    @Id
    @Column(name = "name", nullable = false, updatable = false)
    @NotEmpty
    @Override
    public String getId() {
        return id;
    }

    @Override
    public void setId(String id) {
        this.id = id;
    }

    @Column(nullable = false)
    @NotEmpty
    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    @Transient
    @Override
    public String getName() {
        return null;
    }
}

File: elements/src/main/kotlin/com/rainmore/cms/domains/core/system/security/SystemProperty.kt
package com.rainmore.cms.domains.core.system.security

import com.rainmore.cms.domains.Domain
import com.rainmore.cms.domains.HasId
import com.rainmore.cms.domains.Nameable
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table
import javax.persistence.Transient
import javax.validation.constraints.NotNull

@Entity(name = "core.system.properties")
@Table(name = "systemProperties")
data class SystemProperty(
        @Id
        @Column(name = "name")
        override var id: String?,

        @NotNull
        @Column
        var data: String? = null
) : Domain, HasId<String>, Nameable {
    constructor(id: String) : this(id, null)

    @Transient
    override val name: String? = id
}